SU2_CFD/src/output/filewriter/CParaviewXMLFileWriter.cpp

```diff
@@ -0,0 +1,613 @@
+
+      /*--- Load up the buffer for writing this rank's vector data. ---*/
+
+      float val = 0.0;
+      for (iPoint = 0; iPoint < myPoint; iPoint++) {
+        for (iDim = 0; iDim < NCOORDS; iDim++) {
+          if (nDim == 2 && iDim == 2) {
+            dataBufferFloat[iPoint*NCOORDS + iDim] = 0.0;
+          } else {
+            val = (float)dataSorter->GetData(VarCounter+iDim,iPoint);
+            dataBufferFloat[iPoint*NCOORDS + iDim] = val;
+          }
+        }
+      }
+
+      WriteDataArray(dataBufferFloat.data(), VTKDatatype::FLOAT32, myPoint*NCOORDS, GlobalPoint*NCOORDS, nPoint_Cum[rank]*NCOORDS);
+
+      VarCounter++;
+
+    } else if (output_variable) {
+
+
+      /*--- For now, create a temp 1D buffer to load up the data for writing.
+       This will be replaced with a derived data type most likely. ---*/
+
+      for (iPoint = 0; iPoint < myPoint; iPoint++) {
+        float val = (float)dataSorter->GetData(VarCounter,iPoint);
+        dataBufferFloat[iPoint] = val;
+      }
+
+      WriteDataArray(dataBufferFloat.data(), VTKDatatype::FLOAT32, myPoint, GlobalPoint, nPoint_Cum[rank]);
+
+      VarCounter++;
+    }
+
+  }
+
+  WriteString("</AppendedData>\n", MASTER_NODE);
+  WriteString("</VTKFile>\n", MASTER_NODE);
+
+#ifdef HAVE_MPI
+  /*--- All ranks close the file after writing. ---*/
+
+  MPI_File_close(&fhw);
+#else
+  fclose(fhw);
+#endif
+
+  /*--- Compute and store the write time. ---*/
+
+#ifndef HAVE_MPI
+  stopTime = su2double(clock())/su2double(CLOCKS_PER_SEC);
+#else
+  stopTime = MPI_Wtime();
+#endif
+  usedTime = stopTime-startTime;
+
+  /*--- Communicate the total file size for the restart ---*/
+
+#ifdef HAVE_MPI
+  su2double my_fileSize = fileSize;
+  SU2_MPI::Allreduce(&my_fileSize, &fileSize, 1,
+                     MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+#endif
+
+  /*--- Compute and store the bandwidth ---*/
+
+  bandwidth = fileSize/(1.0e6)/usedTime;
+
+  /*--- Delete the offset counters that we needed for MPI IO. ---*/
+
+  delete [] nElem_Snd;        delete [] nElem_Cum;
+  delete [] nElemStorage_Snd; delete [] nElemStorage_Cum;
+  delete [] nPoint_Snd;       delete [] nPoint_Cum;
+
+}
+
+
+void CParaviewXMLFileWriter::WriteString(std::string str, int rankOut){
+
+#ifdef HAVE_MPI
+  /*--- Reset the file view before writing the next ASCII line for cells. ---*/
+
+  MPI_File_set_view(fhw, 0, MPI_BYTE, MPI_BYTE,
+                    "native", MPI_INFO_NULL);
+
+  if (SU2_MPI::GetRank() == rankOut)
+    MPI_File_write_at(fhw, disp, str.c_str(), int(strlen(str.c_str())),
+                      MPI_CHAR, MPI_STATUS_IGNORE);
+  disp += strlen( str.c_str())*sizeof(char);
+  fileSize += sizeof(char)*strlen( str.c_str());
+#else
+  char str_buf[255];
+  strcpy(str_buf, str.c_str());
+  fwrite(str_buf, sizeof(char), strlen(str_buf), fhw);
+  fileSize += sizeof(char)*strlen(str_buf);
+#endif
+
+}
+
+void CParaviewXMLFileWriter::WriteDataArray(void* data, VTKDatatype type, unsigned long arraySize,
```

Comment:
I like how you broke these methods calling the MPI IO out. Seems to me that some of them could potentially be moved up to the parent class to save duplication. We use similar patterns for the writing of contiguous chunks of binary data (restart, ParaView writers, etc.). Is this possible without too much extra work?